metron-deployment/extra_modules/ambari_service_state.py

```diff
@@ -0,0 +1,352 @@
+  ambari_cluster_state:
+    host: localhost
+    port: 8080
+    username: admin
+    password: admin
+    cluster_name: my_cluster
+    cluster_state: stopped
+    wait_for_complete: True
+- name: Delete the ambari cluster
+  ambari_cluster_state:
+    host: localhost
+    port: 8080
+    username: admin
+    password: admin
+    cluster_name: my_cluster
+    cluster_state: absent
+'''
+
+RETURN = '''
+results:
+    description: The content of the requests object returned from the RESTful call
+    returned: success
+    type: string
+'''
+
+__author__ = 'apachemetron'
+
+import json
+
+try:
+    import requests
+except ImportError:
+    REQUESTS_FOUND = False
+else:
+    REQUESTS_FOUND = True
+
+
+def main():
+
+    argument_spec = dict(
+        host=dict(type='str', default=None, required=True),
+        port=dict(type='int', default=None, required=True),
+        username=dict(type='str', default=None, required=True),
+        password=dict(type='str', default=None, required=True),
+        cluster_name=dict(type='str', default=None, required=True),
+        state=dict(type='str', default=None, required=True,
+                           choices=['started', 'stopped', 'deleted']),
+        service_name=dict(type='str', required=False),
+        component_name=dict(type='str', default=None, required=False),
+        component_host=dict(type='str', default=None, required=False),
+        wait_for_complete=dict(default=False, required=False, choices=BOOLEANS),
+    )
+
+    required_together = ['component_name', 'component_host']
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        required_together=required_together
+    )
+
+    if not REQUESTS_FOUND:
+        module.fail_json(
+            msg='requests library is required for this module')
+
+    p = module.params
+
+    host = p.get('host')
+    port = p.get('port')
+    username = p.get('password')
+    password = p.get('password')
+    cluster_name = p.get('cluster_name')
+    state = p.get('state')
+    service_name = p.get('service_name')
+    component_name = p.get('component_name')
+    component_host = p.get('component_host')
+    wait_for_complete = p.get('wait_for_complete')
+    component_mode = False
+    ambari_url = 'http://{0}:{1}'.format(host, port)
+
+    if component_name:
+        component_mode = True
+
+    try:
+        if not cluster_exists(ambari_url, username, password, cluster_name):
+            module.fail_json(msg="Cluster name {0} does not exist".format(cluster_name))
+
+        if state in ['started', 'stopped', 'installed']:
+            desired_state = ''
+
+            if state == 'started':
+                desired_state = 'STARTED'
+            elif state in ['stopped','installed']:
+                desired_state = 'INSTALLED'
+
+            if component_mode:
+                if desired_state == 'INSTALLED':
+                    if(can_add_component(ambari_url, username, password, cluster_name, component_name, component_host)):
```

Comment:
No, because of 'admin:admin'. I'll look at it again, but it should be fine once the username = password thing is fixed.